File: test_resources/swift/rpc_sync_methods.swift
struct ExamplesSyncRpc {
    private static let scopeId = "4de616f8-12c5-4d2c-8d48-9c5fb038991f"

    static func printHelloWorld() {
        TechPawsBuffersRpc.rpc(
            scopeId: scopeId,
            methodId: 0
        ) { runtime in
            runtime.callRpc()
        }
    }

    static func helloWorld() -> String {
        TechPawsBuffersRpc.rpc(
            scopeId: scopeId,
            methodId: 1
        ) { runtime in
            runtime.callRpc()

            return runtime.readResult() { bytesReader in
                return String.readFromBuffers(bytesReader)
            }
        }
    }

    static func sayHello(
        firstName: String,
        lastName: String
    ) -> String {
        TechPawsBuffersRpc.rpc(
            scopeId: scopeId,
            methodId: 2
        ) { runtime in
            runtime.writeArgs() { bytesWriter in
                firstName.writeToBuffers(bytesWriter)
                lastName.writeToBuffers(bytesWriter)
            }
            runtime.callRpc()

            return runtime.readResult() { bytesReader in
                return String.readFromBuffers(bytesReader)
            }
        }
    }

    static func sum(
        a: Int32,
        b: Float,
        c: Double
    ) -> Double {
        TechPawsBuffersRpc.rpc(
            scopeId: scopeId,
            methodId: 3
        ) { runtime in
            runtime.writeArgs() { bytesWriter in
                bytesWriter.writeInt32(a)
                bytesWriter.writeFloat(b)
                bytesWriter.writeDouble(c)
            }
            runtime.callRpc()

            return runtime.readResult() { bytesReader in
                return bytesReader.readDouble()
            }
        }
    }
}

File: test_resources/kotlin/rpc_sync_methods.kt
object ExamplesSyncRpc {
    private const val SCOPE_ID = "4de616f8-12c5-4d2c-8d48-9c5fb038991f"

    fun printHelloWorld() {
        TechPawsBuffersRpcRuntime.rpc(
            scopeId = SCOPE_ID,
            methodId = 0U,
        ) { runtime ->
            runtime.callRpc()
        }
    }

    fun helloWorld(): String {
        return TechPawsBuffersRpcRuntime.rpc(
            scopeId = SCOPE_ID,
            methodId = 1U,
        ) { runtime ->
            runtime.callRpc()

            runtime.readResult { reader ->
                String.readFromBuffers(reader)
            }
        }
    }

    fun sayHello(
        firstName: String,
        lastName: String,
    ): String {
        return TechPawsBuffersRpcRuntime.rpc(
            scopeId = SCOPE_ID,
            methodId = 2U,
        ) { runtime ->
            runtime.writeArgs { writer ->
                firstName.writeToBuffers(writer)
                lastName.writeToBuffers(writer)
            }
            runtime.callRpc()

            runtime.readResult { reader ->
                String.readFromBuffers(reader)
            }
        }
    }

    fun sum(
        a: Int,
        b: Float,
        c: Double,
    ): Double {
        return TechPawsBuffersRpcRuntime.rpc(
            scopeId = SCOPE_ID,
            methodId = 3U,
        ) { runtime ->
            runtime.writeArgs { writer ->
                a.writeToBuffers(writer)
                b.writeToBuffers(writer)
                c.writeToBuffers(writer)
            }
            runtime.callRpc()

            runtime.readResult { reader ->
                Double.readFromBuffers(reader)
            }
        }
    }
}
